matchbox-server/src/test/java/ch/ahdis/matchbox/test/MatchboxApiR5Test.java

```diff
@@ -0,0 +1,226 @@
+package ch.ahdis.matchbox.test;
+
+import ca.uhn.fhir.context.BaseRuntimeChildDefinition;
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.context.FhirVersionEnum;
+import ca.uhn.fhir.context.RuntimeResourceDefinition;
+import ca.uhn.fhir.jpa.starter.Application;
+import org.apache.commons.io.FileUtils;
+import org.hl7.fhir.instance.model.api.*;
+import org.hl7.fhir.r4.model.OperationOutcome;
+import org.hl7.fhir.r4.model.OperationOutcome.IssueSeverity;
+import org.hl7.fhir.r4.model.OperationOutcome.OperationOutcomeIssueComponent;
+import org.hl7.fhir.r4.model.Parameters;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
+@ContextConfiguration(classes = { Application.class })
+@ActiveProfiles("test-r5")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class MatchboxApiR5Test {
+
+  static public int getValidationFailures(OperationOutcome outcome) {
+    int fails = 0;
+    if (outcome != null && outcome.getIssue() != null) {
+      for (OperationOutcomeIssueComponent issue : outcome.getIssue()) {
+        if (IssueSeverity.FATAL == issue.getSeverity()) {
+          ++fails;
+        }
+        if (IssueSeverity.ERROR == issue.getSeverity()) {
+          ++fails;
+        }
+      }
+    }
+    return fails;
+  }
+
+  private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(MatchboxApiR5Test.class);
+
+  private String targetServer = "http://localhost:8082/matchboxv3/fhir";
+
+  @BeforeAll
+  void waitUntilStartup() throws Exception {
+    Path dir = Paths.get("database");
+    if (Files.exists(dir)) {
+      for (Path file : Files.list(dir).collect(Collectors.toList())) {
+        if (Files.isRegularFile(file)) {
+          Files.delete(file);
+        }
+      }
+    }
+    Thread.sleep(10000); // give the server some time to start up
+    FhirContext contextR4 = FhirVersionEnum.R4.newContext();
```

Comment:
should we switch here not to an R5 context?

Reply:
No, the exchange is still done in FHIR R4 here. Do you want to to also initiate Matchbox in R5?